Source/JavaScriptCore/runtime/CommonSlowPaths.cpp

```diff
@@ -682,11 +682,15 @@ JSC_DEFINE_COMMON_SLOW_PATH(slow_path_bitnot)
 {
     BEGIN();
     auto bytecode = pc->as<OpBitnot>();
+    auto& profile = codeBlock->unlinkedCodeBlock()->unaryArithProfile(bytecode.m_profileIndex);
     auto operand = GET_C(bytecode.m_operand).jsValue();
 
     auto result = jsBitwiseNot(globalObject, operand);
     CHECK_EXCEPTION();
-    RETURN_PROFILED(result);
+    RETURN_WITH_PROFILING(result, {
+        profile.argSawNonNumber();
```

Comment:
Maybe it's irrelevant, but I don't think this is correct. It's not int32, but could still be a number, right?

Reply:
Nice, fixed.